src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageSectorSelection/ImageSectorSelection.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import {action, computed, observable} from 'mobx';
+import DimensionsConverter from './DimensionsConverter';
+import React from 'react';
+import RectangleSelection from '../RectangleSelection';
+import RoundingNormalizer from '../RectangleSelection/dataNormalizers/RoundingNormalizer';
+import type {SelectionData} from '../RectangleSelection/types';
+import log from 'loglevel';
+import {observer} from 'mobx-react';
+import selectionStyles from './imageSectorSelection.scss';
+
+@observer
+export default class ImageSectorSelection extends React.PureComponent {
+    props: {
+        /** Determines the position at which the selection box is rendered at the beginning. */
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        imageSrc: string,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+
+    @observable imageLoaded = false;
+    @observable containerWidth: number;
+    @observable containerHeight: number;
+
+    @action componentWillMount() {
+        this.image = new Image();
+        this.image.onload = action(() => this.imageLoaded = true);
+        this.image.onerror = () => log.error('Failed to preload image');
```

Comment:
`throw new Error` bubbles to the next catch block or causes the application to stop completely. However, because the error would be thrown in an async method, only the execution of this async method would be aborted.